src/common/com/intellij/plugins/haxe/model/HaxeStdTypesFileModel.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2018 Ilya Malanin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intellij.plugins.haxe.model;
+
+import com.intellij.plugins.haxe.lang.psi.HaxeFile;
+import org.jetbrains.annotations.NotNull;
+
+public class HaxeStdTypesFileModel extends HaxeFileModel {
+
+  public HaxeStdTypesFileModel(@NotNull HaxeFile file) {
+    super(file);
+  }
+
+  @Override
+  protected boolean isReferencingCurrentFile(FullyQualifiedInfo info) {
+    return (info.packagePath == null || info.packagePath.isEmpty()) && (info.fileName == null || info.fileName.isEmpty());
```

Comment:
Isn't the current file still being referenced if the FQI resolves to the current file?  This only returns true if the path or filename are empty.

Reply:
Given that this is the StdTypesModel, I think I am missing something...

Reply:
StdTypes - reserved haxe module, that is exposing classes into it as Top Level classes, without package. Like Null, Dynamic and so on. This model will be used only for SDK root, and only for top-level package. Check `HaxeStdPackageModel#resolve`.

Reply:
All right.  I think you still need to check for a qualifier, though, because `import Std;` can also be used in imports (although I would guess that nobody does that).

Reply:
`Std` not equals `StdTypes`. 
`Std` is just a top-level extern class, it will be resolved as expected, by sdk source root within top-level package (same as `UInt`, `Lambda`, `Enum` and so on), even if you'll declare import explicitly.

StdTypes will be resolved (but will be skipped anyway, because it has no Std class declared in it) only in case if Std class for some reason will not be found before.